### PR TITLE
Update quarkchain-web3.js

### PR DESCRIPTION
**Pull request by superman285**
- add support for web3 instance constructed by 1.x version Web3 when using injectWeb3
- update the demos

### CODE REVIEW

File: examples/browser/dapp.html [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
-  <script src="https://cdn.jsdelivr.net/npm/quarkchain-web3@2.0.0/dist/quarkchain-web3.js"></script>
+  <script src="https://cdn.jsdelivr.net/npm/quarkchain-web3@2.1.0/dist/quarkchain-web3.js"></script>
 </head>
 
 <body>
   <h1>Send Tx</h1>
   <form id="sendTxForm">
     <br> To:
     <br>
-    <input type="text" name="to" value="0x000000000000000000000000000000000000000000000001">
```
Comment (ninjaahhh) [inline examples/browser/dapp.html]: why change this default value?

**Review by ninjaahhh — COMMENTED** [review]
why do we need a lib directory for dependencies like `cookiejar` and others? can we use npm packages?

File: index.js [hunk 2]
```diff
@@ -37,9 +37,21 @@ let Web3;
 if (typeof window === 'undefined') {
   Web3 = require("web3"); // eslint-disable-line
 } else {
-  Web3 = window.Web3; // eslint-disable-line
+  if (!window.Web3) {
+    Web3 = require('web3')
+  }else {
```
Comment (ninjaahhh) [inline index.js]: please use eslint for code format and linting

Comment (superman285) [reply index.js]: 👌OK, I will use eslint

File: examples/browser/dapp.html (hunk 1 above)
Comment (superman285) [reply examples/browser/dapp.html]: It's because 'xhr2-cookies' library used by Web3.providers.HttpProvider can not work on browser~

Comment (superman285) [reply examples/browser/dapp.html]: > why change this default value?

Changed back the default value

File: index.js [hunk 3]
```diff
@@ -34,11 +34,21 @@ import ethUtil from 'ethereumjs-util';
 import Transaction from './quarkchain-ethereum-tx';
 
 let Web3;
+/* eslint-disable */
 if (typeof window === 'undefined') {
-  Web3 = require("web3"); // eslint-disable-line
+  Web3 = require('web3');
```
Comment (ninjaahhh) [inline index.js]: condition from line 40 to 44 can be combined

File: index.js [hunk 4]
```diff
@@ -198,7 +208,15 @@ export default {
     // Args:
     //     web3in: web3 instance
     //     jrpcUrl: QuarkChain JSON RPC endpoint (e.g., http://localhost:38391)
-    const web3http = new Web3(new Web3.providers.HttpProvider(jrpcUrl));
+    const web3http = new Web3(new Web3HttpProvider(jrpcUrl));
+
+    let tmpWeb3in;
```
Comment (ninjaahhh) [inline index.js]: seems only `web3in.eth` is affected other than `metaMaskSignTyped`. suggest:

```javascript
let web3eth;
if (web3in.version && !web3in.version.api) {
    // web3in version 1.x
    web3eth = (new Web3(web3in.currentProvider)).eth;
} else {
    web3eth = web3in.eth;
}
```
for `metaMaskSignTyped`, you can get rid off its argument `tmpweb3in` by directly passing `from` address along with  `web3in`

Comment (superman285) [reply index.js]: Got it!

File: index.js (hunk 3 above)
Comment (superman285) [reply index.js]: done~

Comment (ninjaahhh) [reply index.js]: the condition of line 38 and line 40 can also be combined, right?

Comment (superman285) [reply index.js]: yep......

File: index.js [hunk 5]
```diff
@@ -34,11 +34,16 @@ import ethUtil from 'ethereumjs-util';
 import Transaction from './quarkchain-ethereum-tx';
 
 let Web3;
-if (typeof window === 'undefined') {
-  Web3 = require("web3"); // eslint-disable-line
+/* eslint-disable */
+if (typeof window === 'undefined' || !window.Web3 || (window.Web3.version && !window.Web3.version.api)) {
+  //no Web3 or window.Web3 version 1.x
```
Comment (ninjaahhh) [inline index.js]: add a space between // and the comment

**Review by ninjaahhh — COMMENTED** [review]
lgtm. fix the nit and I'll approve

File: index.js (hunk 5 above)
Comment (superman285) [reply index.js]: fix spaced-comment

**Review by ninjaahhh — APPROVED** [review]

**Pull request merged**